src/dune_rules/dialect.ml

```diff
@@ -88,17 +89,23 @@ let decode =
   fields
     (let+ name = field "name" string
      and+ impl = field "implementation" (fields (kind Ml_kind.Impl))
-     and+ intf = field "interface" (fields (kind Ml_kind.Intf)) in
-     { name; file_kinds = Ml_kind.Dict.make ~intf ~impl })
+     and+ intf = field_o "interface" (fields (kind Ml_kind.Intf)) in
```

Comment:
I've added a TODO placeholder for now, do you know what this should look like?

Reply:
You can start by writing a test for this: with `(lang dune 3.8)` this should fail with an error, and accepted with `(lang dune 3.9)`.

In terms of implementation:
- add `and+ version = Dune_lang.Syntax.get_exn Stanza.syntax` to get the declared version
- add something along the lines of this after the parsing code:
```ocaml
let interface_optional_since = 3,9 in
if version < interface_optional_since && Option.is_none interface then
  Syntax.Error.since loc Stanza.syntax interface_optional_since
    ~what:"omitting (interface) in dialects"
```

Reply:
Thanks! done.